splinepy/helpme/fit.py

```diff
@@ -231,6 +223,76 @@ def solve_for_control_points(
     return residual
 
 
+def _validate_specifications(n_query, degree, n_control_points, knot_vector):
+    """Given main values of fitting, validates degree and n_control_points.
+    If there's mismatch between expected value and given value, values will be
+    overwritten."""
+    # sanity checks for given values
+    if degree is not None and knot_vector is not None:
+        expected_ncps = len(knot_vector) - degree - 1
+        if n_control_points is not None and n_control_points != expected_ncps:
+            _log.error(
+                f"n_control_points should be {expected_ncps}. Overwriting."
+            )
+
+        n_control_points = expected_ncps
+
+    if knot_vector is not None and n_control_points is not None:
+        expected_degree = len(knot_vector) - n_control_points - 1
+        if degree is not None and degree != expected_degree:
+            _log.error(f"degree should be {expected_degree}. Overwriting.")
+
+        degree = expected_degree
+
+    # we need degree
+    if degree is None:
+        raise ValueError(
+            "Not enough input to determine degree. Please set degree."
+        )
+
+    # no n_control point -> same as query and this will be interpolation
+    if n_control_points is None:
+        n_control_points = n_query
```

Comment:
Does it make sense to inform the user here via a warning or a log message what value is chosen for `n_control_points` under the hood?

Reply:
Added debug log